hpx/lcos/dist_object.hpp

```diff
@@ -0,0 +1,566 @@
+
+    private:
+        data_type data_;
+    };
+
+    template <typename T>
+    class dist_object_part<T&>
+      : public hpx::components::locking_hook<
+            hpx::components::component_base<dist_object_part<T&>>>
+    {
+    public:
+        typedef T& data_type;
+        dist_object_part() {}
+
+        dist_object_part(data_type data)
+          : data_(data)
+        {
+        }
+
+        data_type operator*()
+        {
+            return data_;
+        }
+
+        data_type operator*() const
+        {
+            return data_;
+        }
+
+        T const* operator->() const
+        {
+            return data_;
+        }
+
+        T* operator->()
+        {
+            return data_;
+        }
+
+        T fetch() const
+        {
+            return data_;
+        }
+
+        HPX_DEFINE_COMPONENT_ACTION(dist_object_part, fetch);
+
+    private:
+        data_type data_;
+    };
+}}}
+
+#define REGISTER_DIST_OBJECT_PART_DECLARATION(type)                            \
+    HPX_REGISTER_ACTION_DECLARATION(                                           \
+        hpx::lcos::server::dist_object_part<type>::fetch_action,               \
+        HPX_PP_CAT(__dist_object_part_fetch_action_, type));
+
+/**/
+
+#define REGISTER_DIST_OBJECT_PART(type)                                        \
+    HPX_REGISTER_ACTION(                                                       \
+        hpx::lcos::server::dist_object_part<type>::fetch_action,               \
+        HPX_PP_CAT(__dist_object_part_fetch_action_, type));                   \
+    typedef ::hpx::components::component<                                      \
+        hpx::lcos::server::dist_object_part<type>>                             \
+        HPX_PP_CAT(__dist_object_part_, type);                                 \
+    HPX_REGISTER_COMPONENT(HPX_PP_CAT(__dist_object_part_, type))              \
+    /**/
+
+namespace hpx { namespace lcos {
+    enum class construction_type
+    {
+        Meta_Object,
+        All_to_All
+    };
+}}
+
+namespace hpx { namespace lcos {
+    class meta_object_server
+      : public hpx::components::locking_hook<
+            hpx::components::component_base<meta_object_server>>
+    {
+    public:
+        meta_object_server()
+          : b(hpx::find_all_localities().size())
+        {
+            servers.resize(hpx::find_all_localities().size());
+        }
+
+        std::vector<hpx::id_type> get_server_list()
+        {
+            if (servers.size() == hpx::find_all_localities().size())
+            {
+                num_sent++;
+                return servers;
+            }
+            std::vector<hpx::id_type> empty;
+            empty.resize(0);
+            return empty;
+        }
+
+        std::vector<hpx::id_type> registration(
+            std::size_t source_loc, hpx::id_type id)
+        {
+            {
+                // TODO: duplicated lock: the namespace has lock guard already
```

Comment:
What is the meaning of this comment? Obsolete?

Reply:
Thanks for attention! Removed now.